nnvm/python/nnvm/frontend/keras.py

```diff
@@ -530,17 +570,23 @@ def from_keras(model):
                 # The one exception is InputLayer.  Changing input variable names after conversion
                 # would confuse users, so we should keep them as far as possible.  Fortunately,
                 # they are named uniquely to input_1, input_2, input_3 ... by default.
-                for pred_idx, pred in zip(node.node_indices, node.inbound_layers):
+                zip_node = zip(node.node_indices, node.tensor_indices, node.inbound_layers)
+                for pred_idx, t_idx, pred in zip_node:
                     if isinstance(pred, keras.engine.InputLayer):
                         sym = symtab.get_var(pred.name, must_contain=True)
                     else:
-                        sym = symtab.get_var(pred.name + ':' + str(pred_idx), must_contain=True)
+                        sym_name = pred.name + ':' + str(pred_idx) + ':' + str(t_idx)
+                        sym = symtab.get_var(sym_name, must_contain=True)
                     insym.append(sym)
 
                 if len(insym) == 1:
                     insym = insym[0]
                 keras_op_to_nnvm(insym, keras_layer, keras_layer.name + ':' + str(my_idx), symtab)
 
-    outsym = [symtab.get_var(layer.name + ':0') for layer in model._output_layers]
+    #model._output_coordinates contains the node name, node_index and tensor index.
+    #search output nodes in symtab using this name
+    outsym = [symtab.get_var(oc[0].name + ":" + str(oc[1]) + ":" + str(oc[2]))
+              for oc in model._output_coordinates]
```

Comment:
Rather than writing the comment, I think we can cleanly explain what we want to do here with code:
```
 outsym = [symtab.get_var(layer.name + ":" + str(node_idx) + ":" + str(tensor_idx))
          for layer, node_idx, tensor_idx in model._output_coordinates]
```

Reply:
Modified the comment as below.
```
    #model._output_coordinates contains out_node(oc[0]), node_index(oc[1]) and tensor index(oc[2]).
    #search all output nodes in symtab using the name made from above values. The out symbols
    #were added to symtab in keras_op_to_nnvm using this name. For multiple outputs, group
    #all these output symbols
    outsym = [symtab.get_var(oc[0].name + ":" + str(oc[1]) + ":" + str(oc[2]))
              for oc in model._output_coordinates]
```

Reply:
So you think writing a comment is better than chainging the code here, right?  It's also fine with me.